sssd_test_framework/utils/authentication.py

```diff
@@ -688,25 +738,46 @@ def password(self, username: str, password: str) -> bool:
         rc, _, _, _ = self.password_with_output(username, password)
         return rc == 0
 
-    def password_expired(self, username: str, password: str, new_password: str) -> bool:
+    def password_expired_with_output(
+        self, username: str, password: str, new_password: str
+    ) -> tuple[int, int, str, str]:
         """
-        SSH to the remote host and authenticate the user with password, expect
-        that the password is expired and change it to the new password.
+        SSH to the remote host and authenticate the user with password, expect that the password
+        is expired and change it to the new password and captures standard output and error.
 
         :param username: Username.
         :type username: str
         :param password: Old, expired user password.
         :type password: str
         :param new_password: New user password.
         :type new_password: str
-        :return: True if authentication and password change was successful, False otherwise.
-        :rtype: bool
+        :return: Tuple containing [except return code, command exit code, stdout, stderr].
+        :rtype: Tuple[int, int, str, str]
         """
         result = self.host.conn.expect_nobody(
             rf"""
+            # Disable debug output
+            exp_internal 0
+
+            proc exitmsg {{ msg code }} {{
+                # Close spawned program, if we are in the prompt
+                catch close
+
+                # Wait for the exit code
+                lassign [wait] pid spawnid os_error_flag rc
+
+                puts ""
+                puts "expect result: $msg"
+                puts "expect exit code: $code"
+                puts "expect spawn exit code: $rc"
+                exit $code
+            }}
+
             # It takes some time to get authentication failure
             set timeout {DEFAULT_AUTHENTICATION_TIMEOUT}
             set prompt "\n.*\[#\$>\] $"
+            log_user 1
+            log_file /tmp/expect.log
```

Comment:
Will this file be automatically deleted when the test ends?

Reply:
Same as previous answer.